File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/shared/updatetask.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum UpdateTaskStatusEnum {
    Queued = "Queued",
    Processing = "Processing",
    Pending = "Pending",
    Success = "Success",
    Stopped = "Stopped",
    Error = "Error"
}


// UpdateTask
/** 
 * A task.
 * 
**/
export class UpdateTask extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=action_type" })
  actionType?: string;

  @SpeakeasyMetadata({ data: "json, name=call_type" })
  callType?: string;

  @SpeakeasyMetadata({ data: "json, name=concurrent_limit" })
  concurrentLimit?: number;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: number;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=object" })
  object?: string;

  @SpeakeasyMetadata({ data: "json, name=object_id" })
  objectId?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: UpdateTaskStatusEnum;

  @SpeakeasyMetadata({ data: "json, name=tags" })
  tags?: string[];

  @SpeakeasyMetadata({ data: "json, name=workflow_id" })
  workflowId?: number;
}

File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/shared/updatetask.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils

class UpdateTaskStatusEnum(str, Enum):
    QUEUED = "Queued"
    PROCESSING = "Processing"
    PENDING = "Pending"
    SUCCESS = "Success"
    STOPPED = "Stopped"
    ERROR = "Error"


@dataclass_json
@dataclasses.dataclass
class UpdateTask:
    r"""UpdateTask
    A task.
    
    """
    
    id: int = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    action_type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('action_type') }})
    call_type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('call_type') }})
    concurrent_limit: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('concurrent_limit') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    object: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('object') }})
    object_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('object_id') }})
    status: Optional[UpdateTaskStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    tags: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tags') }})
    workflow_id: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('workflow_id') }})
    
